### ISSUE TITLE
Fix `allowEmptyInput` option ignored in configuration object regression

### ISSUE THREAD

**Issue by markgomez** [stylelint/stylelint#7440]
### What minimal example or steps are needed to reproduce the bug?

Create a project:
```bash
mkdir foo && cd foo
```
```bash
touch package.json
```
```js
{
  "private": true,
  "scripts": {
    "pass": "stylelint \"src/**/*.css\" --allow-empty-input",
    "fail": "stylelint \"src/**/*.css\""
  },
  "devDependencies": {
    "stylelint": "16.1.0",
    "stylelint-config-standard": "36.0.0"
  }
}
```
```bash
npm i
```

Add a configuration file:
```bash
touch .stylelintrc.json
```
```js
{
  "extends": ["stylelint-config-standard"],
  "allowEmptyInput": true
}
```

Run the `pass` script which includes the `--allow-empty-input` CLI option (this should succeed)
```bash
npm run pass
# OK
```

Run the `fail` script which omits the `--allow-empty-input` CLI option (the `allowEmptyInput` option in the configuration file is expected to allow the script to succeed, but an error occurs instead)
```bash
npm run fail
# Error: No files matching the pattern "src/**/*.css" were found.
```

### What minimal configuration is needed to reproduce the bug?

```js
{
  "extends": ["stylelint-config-standard"],
  "allowEmptyInput": true
}
```

### How did you run Stylelint?

Via NPM scripts in `package.json`.

### Which Stylelint-related dependencies are you using?

- `stylelint@16.1.0`
- `stylelint-config-standard@36.0.0`

### What did you expect to happen?

The `allowEmptyInput` option in the configuration file should allow the lint to succeed despite omitting the `--allow-empty-input` CLI option.

### What actually happened?

The `allowEmptyInput` option in the configuration file seems to be ignored. Everything works when downgrading to version 15, so this might be a version 16 bug. This issue was found and fixed in the previous version (#6926). I'm not sure if this was maybe a breaking change, but I wasn't able to find any mention of it when going over [the version 16 migration guide](https://github.com/stylelint/stylelint/blob/16.0.0/docs/migration-guide/to-16.md).

**Comment by ybiquitous** [comment]
@markgomez Thanks for the report with details. As you mentioned, this bug is a regression in Stylelint 16.0.0 due to the CLI module refactoring in #7129. Relevant code:

https://github.com/stylelint/stylelint/blob/804bb24c75248aba55b009994e4bfb561593a990/lib/cli.mjs#L477-L479

https://github.com/stylelint/stylelint/blob/804bb24c75248aba55b009994e4bfb561593a990/lib/standalone.mjs#L258

Since 16.0.0, `cli.mjs` has passed `options.allowEmptyInput === false` instead of `undefined` into `standalone.mjs`, although `standalone.js` still has expected `undefined`.

I'll try fixing this bug soon.
